Cannot build on OpenBSD

OpenBSD does not support TFO, hence this error:

```
error: Doesn't support TFO on the current platform
  --> /home/builder/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-tfo-0.2.2/src/sys/unix/bsd/mod.rs:14:9
   |
14 |         compile_error!("Doesn't support TFO on the current platform");
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0432]: unresolved import `crate::sys::set_tcp_fastopen`
  --> /home/builder/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-tfo-0.2.2/src/listener.rs:15:32
   |
15 | use crate::{stream::TfoStream, sys::set_tcp_fastopen};
   |                                ^^^^^^^^^^^^^^^^^^^^^ no `set_tcp_fastopen` in `sys`

error[E0432]: unresolved import `crate::sys::TcpStream`
  --> /home/builder/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-tfo-0.2.2/src/stream.rs:21:5
   |
21 | use crate::sys::TcpStream as SysTcpStream;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `TcpStream` in `sys`
   |
help: consider importing one of these items instead
   |
21 | use std::net::TcpStream as SysTcpStream;
   |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
21 | use tokio::net::TcpStream as SysTcpStream;
   |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

For more information about this error, try `rustc --explain E0432`.
error: could not compile `tokio-tfo` (lib) due to 3 previous errors
warning: build failed, waiting for other jobs to finish...
```

--- a/crates/shadowsocks/Cargo.toml
+++ b/crates/shadowsocks/Cargo.toml
@@ -81,7 +81,6 @@ tokio = { version = "1.9.0", features = [
     "sync",
     "time",
 ] }
-tokio-tfo = "0.2.0"
 
 hickory-resolver = { version = "0.24", optional = true }
 arc-swap = { version = "1.7", optional = true }
@@ -95,6 +94,9 @@ shadowsocks-crypto = { version = "0.5.4", features = ["ring"] }
 
 [target.'cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))'.dependencies]
 shadowsocks-crypto = { version = "0.5.4", features = [] }
+
+[target.'cfg(any(windows, target_os = "linux", target_os = "android", target_os = "freebsd", target_os = "macos", target_os = "ios", target_os = "watchos", target_os = "tvos"))'.dependencies]
+tokio-tfo = "0.2.0"
 
 [target.'cfg(windows)'.dependencies]
 windows-sys = { version = "0.52", features = [

